### PR TITLE
Users CRUD REST API Endpoints

### PR DESCRIPTION
**Pull request by shashisp**

### CODE REVIEW

**Comment by adamstep** [comment]
Hi @shashisp ,

I've been attempting to set up your teamapp backend, but I'm running into some errors. In particular, I'm stuck on running "python manage.py migrate" according to the README. I get the following error:

```
  File "/Users/adam/shashisp/teamapp-backend/env/lib/python3.7/site-packages/django/db/backends/mysql/base.py", line 15, in <module>
    import MySQLdb as Database
ModuleNotFoundError: No module named 'MySQLdb'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "manage.py", line 21, in <module>
    main()
  File "manage.py", line 17, in main
    execute_from_command_line(sys.argv)
  File "/Users/adam/shashisp/teamapp-backend/env/lib/python3.7/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/Users/adam/shashisp/teamapp-backend/env/lib/python3.7/site-packages/django/core/management/__init__.py", line 357, in execute
    django.setup()
  File "/Users/adam/shashisp/teamapp-backend/env/lib/python3.7/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/Users/adam/shashisp/teamapp-backend/env/lib/python3.7/site-packages/django/apps/registry.py", line 114, in populate
    app_config.import_models()
  File "/Users/adam/shashisp/teamapp-backend/env/lib/python3.7/site-packages/django/apps/config.py", line 211, in import_models
    self.models_module = import_module(models_module_name)
  File "/Users/adam/shashisp/teamapp-backend/env/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/adam/shashisp/teamapp-backend/env/lib/python3.7/site-packages/django/contrib/auth/models.py", line 2, in <module>
    from django.contrib.auth.base_user import AbstractBaseUser, BaseUserManager
  File "/Users/adam/shashisp/teamapp-backend/env/lib/python3.7/site-packages/django/contrib/auth/base_user.py", line 47, in <module>
    class AbstractBaseUser(models.Model):
  File "/Users/adam/shashisp/teamapp-backend/env/lib/python3.7/site-packages/django/db/models/base.py", line 117, in __new__
    new_class.add_to_class('_meta', Options(meta, app_label))
  File "/Users/adam/shashisp/teamapp-backend/env/lib/python3.7/site-packages/django/db/models/base.py", line 321, in add_to_class
    value.contribute_to_class(cls, name)
  File "/Users/adam/shashisp/teamapp-backend/env/lib/python3.7/site-packages/django/db/models/options.py", line 204, in contribute_to_class
    self.db_table = truncate_name(self.db_table, connection.ops.max_name_length())
  File "/Users/adam/shashisp/teamapp-backend/env/lib/python3.7/site-packages/django/db/__init__.py", line 28, in __getattr__
    return getattr(connections[DEFAULT_DB_ALIAS], item)
  File "/Users/adam/shashisp/teamapp-backend/env/lib/python3.7/site-packages/django/db/utils.py", line 201, in __getitem__
    backend = load_backend(db['ENGINE'])
  File "/Users/adam/shashisp/teamapp-backend/env/lib/python3.7/site-packages/django/db/utils.py", line 110, in load_backend
    return import_module('%s.base' % backend_name)
  File "/Users/adam/shashisp/teamapp-backend/env/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/Users/adam/shashisp/teamapp-backend/env/lib/python3.7/site-packages/django/db/backends/mysql/base.py", line 20, in <module>
    ) from err
django.core.exceptions.ImproperlyConfigured: Error loading MySQLdb module.
Did you install mysqlclient?
```

I think maybe you're missing something in `requirements.txt`?

**Comment by shashisp** [comment]
@adamstep yes, I used the `mysqlclient`, my bad, didn't notice that one, let me update the requirements.txt

**Comment by adamstep** [comment]
@shashisp apologies, I didn't get anotification for your update so I didn't check back. I'll try installing the project again to review today.

**Comment by shashisp** [comment]
I'm sorry. I should've emailed after the changes.

File: users/models.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from django.db import models
+from django.contrib.auth.models import AbstractUser
+
+
+class CustomUser(AbstractUser):
+    phone = models.CharField(max_length=15, blank=True)
```
Comment (adamstep) [inline users/models.py]: This field allows all sorts of non-phone number inputs which could cause problems in the application. Can you use a field with stricter validation? There may be som Django libraries that implement a phone number field.

**Comment by adamstep** [comment]
@shashisp I was able to test, the functionality works well. See my one request above.

File: users/models.py (hunk 1 above)
Comment (shashisp) [reply users/models.py]: @adamstep  sure, among all other libraries,  this one looks like an active project and also has `PhoneNumberField ` field at serializer level https://github.com/stefanfoulis/django-phonenumber-field, updated with the PR with a phone number field.

**Pull request merged**